plugins/circleci/src/pages/BuildsPage/lib/CITable/CITable.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { FC } from 'react';
+import { Link, Typography, Box, IconButton } from '@material-ui/core';
+import { Replay as RetryIcon, GitHub as GithubIcon } from '@material-ui/icons';
+import { Link as RouterLink } from 'react-router-dom';
+import {
+  StatusFailed,
+  StatusOK,
+  StatusPending,
+  StatusNA,
+  StatusRunning,
+  Table,
+  TableColumn,
+} from '@backstage/core';
+
+export type CITableBuildInfo = {
+  id: string;
+  buildName: string;
+  buildUrl?: string;
+  source: {
+    branchName: string;
+    commit: {
+      hash: string;
+      url: string;
+    };
+  };
+  status: string;
+  tests?: {
+    total: number;
+    passed: number;
+    skipped: number;
+    failed: number;
+    testUrl: string; // fixme better name
+  };
+  onRetryClick: () => void;
+};
+
+// retried, canceled, infrastructure_fail, timedout, not_run, running, failed, queued, scheduled, not_running, no_tests, fixed, success
+const getStatusComponent = (status: string | undefined = '') => {
+  switch (status.toLowerCase()) {
+    case 'queued':
+    case 'scheduled':
+      return <StatusPending />;
+    case 'running':
+      return <StatusRunning />;
+    case 'failed':
+      return <StatusFailed />;
+    case 'success':
+      return <StatusOK />;
+    case 'canceled':
+    default:
+      return <StatusNA />;
+  }
+};
+
+const generatedColumns: TableColumn[] = [
+  {
+    title: 'ID',
+    field: 'id',
+    type: 'numeric',
+    width: '80px',
+  },
+  {
+    title: 'Build',
+    field: 'buildName',
+    highlight: true,
+    render: (row: Partial<CITableBuildInfo>) => (
+      <Link component={RouterLink} to={`/circleci/build/${row.id}`}>
+        {row.buildName}
+      </Link>
+    ),
+  },
+  {
+    title: 'Source',
+    render: (row: Partial<CITableBuildInfo>) => (
+      <>
+        {row.source?.branchName}
+        <br />
```

Comment:
I thought using `<br>` was frowned upon 😄

Reply:
Makes sense. I've changed this to <p/>s. It adds a little bit of additional spacing but I think it reads better.